lib/rex/proto/http/web_socket.rb

```diff
@@ -1,428 +1,449 @@
 # -*- coding: binary -*-
+
 require 'bindata'
 
 module Rex
-module Proto
-module Http
-module WebSocket
-
-class WebSocketError < StandardError
-end
-
-class ConnectionError < WebSocketError
-  def initialize(msg: 'The WebSocket connection failed', http_response: nil)
-    @message = msg
-    @http_response = http_response
-  end
-
-  attr_accessor :message, :http_response
-  alias :to_s :message
-end
-
-# This defines the interface that the standard socket is extended with to provide WebSocket functionality. It should be
-# used on a socket when the server has already successfully handled a WebSocket upgrade request.
-module Interface
-  #
-  # A channel object that allows reading and writing either text or binary data directly to the remote peer.
-  #
-  class Channel
-    include Rex::Post::Channel::StreamAbstraction
-
-    module SocketInterface
-      include Rex::Post::Channel::SocketAbstraction::SocketInterface
-
-      def type?
-        'tcp'
-      end
-    end
-
-    # The socket parameters describing the underlying connection.
-    # @!attribute [r] params
-    #   @return [Rex::Socket::Parameters]
-    attr_reader :params
-
-    # @param [WebSocket::Interface] websocket the WebSocket that this channel is being opened on
-    # @param [nil, Symbol] read_type the data type(s) to read from the WebSocket, one of :binary, :text or nil (for both
-    #   binary and text)
-    # @param [Symbol] write_type the data type to write to the WebSocket
-    def initialize(websocket, read_type: nil, write_type: :binary)
-      initialize_abstraction
-
-      # a read type of nil will handle both binary and text frames that are received
-      raise ArgumentError, 'read_type must be nil, :binary or :text' unless [nil, :binary, :text].include?(read_type)
-      raise ArgumentError, 'write_type must be :binary or :text' unless [:binary, :text].include?(write_type)
-      @websocket = websocket
-      @read_type = read_type
-      @write_type = write_type
-      @mutex = Mutex.new
-
-      # beware of: https://github.com/rapid7/rex-socket/issues/32
-      _, localhost, localport = websocket.getlocalname
-      _, peerhost, peerport = Socket.from_sockaddr(websocket.getpeername)
-      @params = Rex::Socket::Parameters.from_hash({
-        'LocalHost' => localhost,
-        'LocalPort' => localport,
-        'PeerHost' => peerhost,
-        'PeerPort' => peerport,
-        'SSL' => websocket.respond_to?(:sslctx) && !websocket.sslctx.nil?
-      })
-
-      @thread = Rex::ThreadFactory.spawn("WebSocketChannel(#{localhost}->#{peerhost})", false) do
-        websocket.wsloop do |data, data_type|
-          next unless @read_type.nil? || data_type == @read_type
-
-          data = on_data_read(data, data_type)
-          next if data.nil?
-
-          rsock.syswrite(data)
+  module Proto
+    module Http
+      module WebSocket
```

Comment:
```suggestion
module Rex::Proto::Http::WebSocket
```

Which might clean up the indentation diff

Reply:
Well I did that, but to keep rubocop happy still, the indentation still changed.